### ISSUE TITLE
ProtoBuf input and output layers

### ISSUE THREAD

**Issue by guimCC** [ultralytics/ultralytics#18870]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I've exported several models as **Tensorflow GraphDef** type, generating the `.pb` files.

Then, I'm able to load the graph from **Tensorflow**, which works perfectly. However, I've extensively searched the documentation (https://docs.ultralytics.com/integrations/tf-graphdef/) and I'm unable to find specific mention on what are the input/ouput point names. Those identifiers are required to perform inference from the graph.

Any help on where to find such documentation? Huge thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @guimCC, thank you for reaching out and sharing your question about exporting models as TensorFlow GraphDef `.pb` files 🚀!

We recommend consulting the [Docs](https://docs.ultralytics.com/integrations/tf-graphdef/) for guidance on TensorFlow integrations. While we aim to make the documentation as comprehensive as possible, some specifics—like input/output layer names—might require exploration based on your custom use case.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate the issue further. For example, sharing relevant sections of your code and details about your export and inference setup would be valuable.

In the meantime, consider verifying the exported graph’s input/output layers using TensorFlow utilities such as `tf.saved_model` tools or third-party visualization tools like [Netron](https://github.com/lutzroeder/netron), which can display the graph structure and corresponding names.

Join the Ultralytics community for additional support and knowledge sharing:
- 💬 [Discord](https://discord.com/invite/ultralytics) for real-time discussions.
- 📚 [Discourse](https://community.ultralytics.com/) for in-depth conversations.
- 📢 [Subreddit](https://reddit.com/r/Ultralytics) to connect with fellow developers.

## Upgrade

First, ensure you’re using the latest `ultralytics` package. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to rule out outdated issues:

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be utilized across several tested environments for development or debugging:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing, ensuring functionality across all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

⚡ This is an automated response, but don’t worry—an Ultralytics engineer will follow up shortly to assist further 😊.

**Comment by Y-T-G** [comment]
You can try these:

https://github.com/ultralytics/ultralytics/blob/9181ff62f539e445f2d992e2e7f08df4d49f5c00/ultralytics/engine/exporter.py#L510-L515

**Comment by guimCC** [comment]
Thanks for your response.

I explored the graphs using https://netron.app/ and found the following tensor names:

- Input: `x:0`
- Output: `Identity:0`

Which worked with no issue.

**Comment by glenn-jocher** [comment]
@guimCC you're welcome! Glad to hear that you resolved it and everything is working smoothly. If you have further questions, feel free to ask.
